rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -430,26 +461,143 @@ def _decode_curve_gauge_events(self, context: DecoderContext) -> DecodingOutput:
+    ) -> list['EvmEvent']:
+        """This method handles post decoding for curve events made through the deposit contracts.
+
+        This is required because certain events are not available when `AddLiquidity` or `RemoveLiquidity` occurs.
+        An example of such is when liquidity is removed, the transfer event of the token removed comes after the
+        `RemoveLiquidity` event.
+        """  # noqa: E501
+        for tx_log in all_logs:
+            if (
+                tx_log.topics[0] in ADD_LIQUIDITY_EVENTS and
+                hex_or_bytes_to_address(tx_log.topics[1]) in CURVE_DEPOSIT_CONTRACTS
+            ):
+                self._handle_zap_contracts_liquidity_addition(
+                    transaction=transaction,
+                    decoded_events=decoded_events,
+                )
+            if (
+                tx_log.topics[0] in REMOVE_LIQUIDITY_EVENTS and
+                hex_or_bytes_to_address(tx_log.topics[1]) in CURVE_DEPOSIT_CONTRACTS
+            ):
+                self._handle_zap_contracts_liquidity_removal(
+                    pool_address=tx_log.address,
+                    transaction=transaction,
+                    decoded_events=decoded_events,
+                )
+
+        return decoded_events  # noop. just return the decoded events.
+
+    def _handle_zap_contracts_liquidity_addition(
+            self,
+            transaction: EvmTransaction,
+            decoded_events: list['EvmEvent'],
+    ) -> None:
+        """
+        This method completes the decoding of liquidity addition made through
+        Curve zap contracts such as e.g. Curve Deposit USDN, etc.
+        """
+        receive_event_idx, deposit_events = None, []
+        for idx, event in enumerate(decoded_events):
+            if (
+                event.event_type == HistoryEventType.DEPOSIT and
+                event.event_subtype == HistoryEventSubType.DEPOSIT_ASSET and
+                event.counterparty == CPT_CURVE
+            ):
+                deposit_events.append(event)
+
+            # search for the event of the wrapped token received after deposit
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.location_label == transaction.from_address and
+                (
+                    event.address in CURVE_DEPOSIT_CONTRACTS or
+                    event.address == ZERO_ADDRESS
+                )
+            ):
+                receive_event_idx = idx
+                crypto_asset = event.asset.resolve_to_crypto_asset()
+                event.event_type = HistoryEventType.RECEIVE
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = CPT_CURVE
+                event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} after depositing in curve pool'  # noqa: E501
```

Comment:
As I also wrote to you in discord and got no response I am doing it here too.

https://discord.com/channels/657906918408585217/1080783409640648765/1092506473491333140

I don't believe there is no way to find the right data here. But I am just tired of this PR being open for so long so if you absolutely can't find it someone else may do as an additional issue in the future.

```suggestion
                event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} after depositing in a curve pool'  # noqa: E501
```